### PR TITLE
129: Update OpenTelemetry

### LINKED ISSUE #129: Use OTLP for Jaeger
Now Jaeger supports generic OTLP, we can switch from Jaeger-specific to a generic OTLP.

### PR DESCRIPTION
**Pull request by ksenia-vazhdaeva**
Closes #129

### CODE REVIEW

File: savant_core/src/telemetry.rs [hunk 1]
```diff
@@ -1,29 +1,37 @@
 use opentelemetry::global;
+use opentelemetry_jaeger_propagator::Propagator;
+use opentelemetry_otlp::WithExportConfig;
 use opentelemetry_sdk::propagation::TraceContextPropagator;
 use opentelemetry_sdk::trace::{Config, TracerProvider};
 use opentelemetry_sdk::Resource;
+use opentelemetry_semantic_conventions::resource::{SERVICE_NAME, SERVICE_NAMESPACE};
+use opentelemetry_stdout::SpanExporter;
 
 pub fn init_jaeger_tracer(service_name: &str, endpoint: &str) {
```
Comment (bwsw) [inline savant_core/src/telemetry.rs:10]: Please rename and update everywhere to let users know that this is a generic tracer, not jaeger: `init_otlp_tracer`.

Also, check it really works and update the test: https://github.com/insight-platform/savant-rs/blob/main/python/pipeline.py

**Review by bwsw — CHANGES_REQUESTED** [review]

File: savant_core/src/telemetry.rs (hunk 1 above)
Comment (bwsw) [reply savant_core/src/telemetry.rs:10]: Please use batch exporter for efficiency reasons.

Comment (ksenia-vazhdaeva) [reply savant_core/src/telemetry.rs:10]: I have checked only with Jaeger and decided to leave context propagation in Jaeger format for backward compatibility.

Do we need to make context propagation format configurable?

 If we want to make this fully configurable, then the configuration will be rather complex starting that both grpc and http should be supported.

About test: do you want to have an integration test that will deploy OLTP backend and check that it works?

Comment (bwsw) [reply savant_core/src/telemetry.rs:10]: I mostly care about auth and compression.
Propagation format: yes, optional
Test: yes

**Pull request merged**